### PR TITLE
Fix regex validator

### LINKED ISSUE #1860: TextToolParameter validation fails when loading up tool.
I noticed this on the latest dev.
If I try to load the datamash tool from the toolbox, I get the following exception.

```
galaxy.web.framework.decorators ERROR 2016-03-04 15:59:11,664 Uncaught exception in exposed API method:
Traceback (most recent call last):
  File "/Users/marius/src/galaxy/lib/galaxy/web/framework/decorators.py", line 259, in decorator
    rval = func( self, trans, *args, **kwargs)
  File "/Users/marius/src/galaxy/lib/galaxy/webapps/galaxy/api/tools.py", line 97, in build
    return tool.to_json(trans, kwd.get('inputs', kwd))
  File "/Users/marius/src/galaxy/lib/galaxy/tools/__init__.py", line 1751, in to_json
    self.populate_state( request_context, self.inputs, params.__dict__, state_inputs, state_errors )
  File "/Users/marius/src/galaxy/lib/galaxy/tools/__init__.py", line 1850, in populate_state
    value, error = check_param( request_context, input, param_value, context )
  File "/Users/marius/src/galaxy/lib/galaxy/tools/parameters/__init__.py", line 106, in check_param
    param.validate( value, trans )
  File "/Users/marius/src/galaxy/lib/galaxy/tools/parameters/basic.py", line 265, in validate
    return super( TextToolParameter, self ).validate( value, trans )
  File "/Users/marius/src/galaxy/lib/galaxy/tools/parameters/basic.py", line 190, in validate
    validator.validate( value, trans )
  File "/Users/marius/src/galaxy/lib/galaxy/tools/parameters/validation.py", line 61, in validate
    if re.match( self.expression, value ) is None:
  File "/Users/marius/src/galaxy/.venv/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or buffer
```

In this instance this is a text type parameter that has a validator associated to it
(https://github.com/galaxyproject/tools-iuc/blob/master/tools/datamash/datamash-ops.xml#L29).
Since on first load the value is `None` the validation fails ...
I guess the validator should be skipped when loading a tool from the toolbox?

### PR DESCRIPTION
**Pull request by guerler**
Should fix #1860.

### CODE REVIEW

File: lib/galaxy/tools/parameters/validation.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def __init__( self, message, expression ):
         self.expression = expression
 
     def validate( self, value, trans=None ):
-        if re.match( self.expression, value ) is None:
+        if re.match( self.expression, value or '' ) is None:
```
Comment (mvdbeek) [inline lib/galaxy/tools/parameters/validation.py:61]: Wouldn't this break the validation of things that are not optional and that cannot be empty?

Comment (mvdbeek) [inline lib/galaxy/tools/parameters/validation.py:61]: Maybe checking [here](https://github.com/galaxyproject/galaxy/blob/02187afb57a3b212c4e9a05b153e117bc8778866/lib/galaxy/tools/parameters/__init__.py#L99) that we are in a tool building context would be better?

update: nevermind, empty values only come in as `""`, so this should be fine!

**Comment by mvdbeek** [comment]
:+1: thanks @guerler

**Comment by guerler** [comment]
Cool, Thanks a lot.

File: lib/galaxy/tools/parameters/validation.py (hunk 1 above)
Comment (guerler) [inline lib/galaxy/tools/parameters/validation.py:61]: I added https://github.com/galaxyproject/galaxy/issues/1867 to track the addition of non-optional text parameters

**Pull request merged**